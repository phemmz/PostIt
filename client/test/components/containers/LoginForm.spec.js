import React from 'react';
import { mount } from 'enzyme';
import { LoginForm } from
  '../../../src/components/layout/login/LoginForm';
import { initialLoginState } from '../../__mockData__/dummyData';

jest.mock('react-google-login');

const setup = () => {
  const props = {
    login: () => Promise.resolve(),
    googleAuthentication: jest.fn()
  };
  const enzymeWrapper = mount(
    <LoginForm {...props} />
  );
  return {
    props,
    enzymeWrapper
  };
};

describe('LoginForm Component', () => {
  const { props, enzymeWrapper } = setup();
  it('should always render a div', () => {
    expect(enzymeWrapper).toMatchSnapshot();
    expect(enzymeWrapper.exists()).toBe(true);
    expect(enzymeWrapper.find('div').exists()).toBe(true);
  });
  it("should contain everything else that gets rendered", () => {
    const divs = enzymeWrapper.find("div");
    const wrappingDiv = divs.first();
    expect(wrappingDiv.children()).toEqual(enzymeWrapper.children());
  });
  it('should have an initial state', () => {
    expect(enzymeWrapper.state()).toEqual(initialLoginState);
  });
  it('should set state on change of username and password input fields',
  () => {
    enzymeWrapper.find('#username').simulate('change', {
      target: { name: 'username', value: 'user1' }
    });
    expect(enzymeWrapper.state('username')).toBe('user1');
    enzymeWrapper.find('#password').simulate('change', {
      target: { name: 'password', value: '123456' }
    });
    expect(enzymeWrapper.state('password')).toBe('123456');
  });
  it('should update the state when login button is clicked', () => {
    const form = enzymeWrapper.find('form').first();
    form.simulate('submit', {
      preventDefault: () => {
      },
      target: [
        {
          name: 'username',
          value: 'user1',
        },
        {
          name: 'password',
          value: '123456',
        }
      ],
    });
    expect(enzymeWrapper.state('username')).toBe('user1');
  });
  it("should only take two props", () => {
    expect(Object.keys(enzymeWrapper.props()).length).toBe(2);
  });
});