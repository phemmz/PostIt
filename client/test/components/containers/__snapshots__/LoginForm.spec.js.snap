// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component should always render a div 1`] = `
<LoginForm
  googleAuthentication={[Function]}
  login={[Function]}
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col s12 m6 offset-s7 offset-m6 login"
      >
        <div
          className="right"
        >
          <t
            buttonText="Sign in with Google"
            className="google-link"
            clientId="484298663558-arrp0kt4u2j2aro9k2bbb2bcffth9fke.apps.googleusercontent.com"
            cookiePolicy="single_host_origin"
            disabledStyle={
              Object {
                "opacity": 0.6,
              }
            }
            fetchBasicProfile={true}
            isSignedIn={false}
            onFailure={[Function]}
            onRequest={[Function]}
            onSuccess={[Function]}
            prompt=""
            responseType="permission"
            scope="profile email"
            tag="button"
            uxMode="popup"
          />
        </div>
        <form
          onSubmit={[Function]}
        >
          <h4
            className="green-text text-darken-4 login-text"
          >
            Login
          </h4>
          <TextFieldGroup
            checkUserExists={[Function]}
            error=""
            field="username"
            htmlFor="username"
            id="username"
            label="Username"
            onChange={[Function]}
            type="text"
            value=""
          >
            <div
              className="input-field col s12"
            >
              <input
                className="form-control"
                id="username"
                name="username"
                onBlur={[Function]}
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
              <label
                htmlFor="username"
              >
                Username
              </label>
            </div>
          </TextFieldGroup>
          <TextFieldGroup
            checkUserExists={[Function]}
            error=""
            field="password"
            htmlFor="password"
            id="password"
            label="Password"
            onChange={[Function]}
            type="password"
            value=""
          >
            <div
              className="input-field col s12"
            >
              <input
                className="form-control"
                id="password"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                required={true}
                type="password"
                value=""
              />
              <label
                htmlFor="password"
              >
                Password
              </label>
            </div>
          </TextFieldGroup>
          <div
            className="center"
          >
            <button
              className="waves-effect waves-light btn"
            >
              Login
            </button>
          </div>
          <div
            className="center reset-link"
          >
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/reset"
            >
              <a
                onClick={[Function]}
                style={Object {}}
              >
                Forgot Password?
              </a>
            </Link>
          </div>
        </form>
      </div>
    </div>
  </div>
</LoginForm>
`;
